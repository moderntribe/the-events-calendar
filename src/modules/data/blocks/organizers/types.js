/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const ADD_CLASSIC_ORGANIZERS = `${ PREFIX_EVENTS_STORE }/ADD_CLASSIC_ORGANIZERS`;
export const REMOVE_CLASSIC_ORGANIZERS = `${ PREFIX_EVENTS_STORE }/REMOVE_CLASSIC_ORGANIZERS`;

export const ADD_BLOCK_ORGANIZER = `${ PREFIX_EVENTS_STORE }/ADD_BLOCK_ORGANIZER`;
export const REMOVE_BLOCK_ORGANIZER = `${ PREFIX_EVENTS_STORE }/REMOVE_BLOCK_ORGANIZER`;
