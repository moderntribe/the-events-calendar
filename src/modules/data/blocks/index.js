/**
 * Internal dependencies
 */
import * as datetime from './datetime';
import subscribe from './subscribers';

export { datetime, subscribe };
export { default, setInitialState } from './reducer';
