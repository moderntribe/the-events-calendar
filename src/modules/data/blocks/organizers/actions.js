/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events/data/blocks/organizers';

export const removeOrganizerInClassic = ( organizer ) => ( {
	type: types.REMOVE_CLASSIC_ORGANIZERS,
	payload: {
		organizer,
	},
} );

export const addOrganizerInClassic = ( organizer ) => ( {
	type: types.ADD_CLASSIC_ORGANIZERS,
	payload: {
		organizer,
	},
} );

export const addOrganizerInBlock = ( id, organizer ) => ( {
	type: types.ADD_BLOCK_ORGANIZER,
	payload: {
		id,
		organizer,
	},
} );

export const removeOrganizerInBlock = ( id, organizer ) => ( {
	type: types.REMOVE_BLOCK_ORGANIZER,
	payload: {
		id,
		organizer,
	},
} );
