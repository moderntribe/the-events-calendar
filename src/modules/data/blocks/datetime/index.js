/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, selectors, sagas, setInitialState, defaultStateToMetaMap };
