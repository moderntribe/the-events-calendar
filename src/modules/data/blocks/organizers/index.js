/**
 * Internal dependencies
 */
import reducer, { setInitialState } from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import subscribe from './subscribers';

export default reducer;
export { types, actions, selectors, setInitialState, subscribe };
