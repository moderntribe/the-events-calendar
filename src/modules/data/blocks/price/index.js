/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import * as utils from './utils';

export default reducer;
export { selectors, actions, types, utils, setInitialState, defaultStateToMetaMap };
