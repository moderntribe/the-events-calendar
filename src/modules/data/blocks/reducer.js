/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime, { setInitialState as datetimeSetInitialState } from './datetime';
import organizers, { setInitialState as organizersSetInitialState } from './organizers';
import price, { setInitialState as priceSetInitialState } from './price';
import website, { setInitialState as websiteSetInitialState } from './website';
import venue, { setInitialState as venueSetInitialState } from './venue';

export const setInitialState = ( data ) => {
	datetimeSetInitialState( data );
	organizersSetInitialState( data );
	priceSetInitialState( data );
	venueSetInitialState( data );
	websiteSetInitialState( data );
};

export default combineReducers( {
	datetime,
	venue,
	organizers,
	price,
	website,
} );
