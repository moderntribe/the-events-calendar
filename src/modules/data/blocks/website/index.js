/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';

export default reducer;
export { selectors, actions, types, setInitialState, defaultStateToMetaMap };
