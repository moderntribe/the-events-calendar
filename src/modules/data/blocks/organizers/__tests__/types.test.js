/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events/data/blocks/organizers';

describe( '[STORE] - Organizers types', () => {
	it( 'Should return the organizers types', () => {
		expect( types.ADD_BLOCK_ORGANIZER ).toMatchSnapshot();
		expect( types.REMOVE_BLOCK_ORGANIZER ).toMatchSnapshot();
		expect( types.ADD_CLASSIC_ORGANIZERS ).toMatchSnapshot();
		expect( types.REMOVE_CLASSIC_ORGANIZERS ).toMatchSnapshot();
	} );
} );
