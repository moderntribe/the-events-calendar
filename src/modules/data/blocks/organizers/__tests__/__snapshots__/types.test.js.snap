// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Organizers types Should return the organizers types 1`] = `"@@MT/EVENTS/ADD_BLOCK_ORGANIZER"`;

exports[`[STORE] - Organizers types Should return the organizers types 2`] = `"@@MT/EVENTS/REMOVE_BLOCK_ORGANIZER"`;

exports[`[STORE] - Organizers types Should return the organizers types 3`] = `"@@MT/EVENTS/ADD_CLASSIC_ORGANIZERS"`;

exports[`[STORE] - Organizers types Should return the organizers types 4`] = `"@@MT/EVENTS/REMOVE_CLASSIC_ORGANIZERS"`;
